src/main/java/seedu/address/logic/commands/edit/EditExamCommand.java

```diff
@@ -0,0 +1,226 @@
+package seedu.address.logic.commands.edit;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.util.CollectionUtil.requireAllNonNull;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_CODE;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_DAY;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_END;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_LINK;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_NAME;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_START;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_VENUE;
+
+import java.util.List;
+import java.util.Optional;
+
+import seedu.address.commons.core.Messages;
+import seedu.address.commons.core.index.Index;
+import seedu.address.commons.util.CollectionUtil;
+import seedu.address.logic.commands.CommandResult;
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.Model;
+import seedu.address.model.module.Link;
+import seedu.address.model.module.ModBookDate;
+import seedu.address.model.module.Module;
+import seedu.address.model.module.ModuleCode;
+import seedu.address.model.module.Timeslot;
+import seedu.address.model.module.Venue;
+import seedu.address.model.module.exam.Exam;
+import seedu.address.model.module.exam.ExamName;
+
+public class EditExamCommand extends EditCommand {
+    public static final String MESSAGE_DUPLICATE_EXAM = "This exam already exists.";
+    public static final String MESSAGE_EDIT_EXAM_SUCCESS = "Edited Exam %s in Module %s";
+    public static final String MESSAGE_NOT_EDITED = "At least one field to edit must be provided.";
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Edits an exam in the Mod book. "
+            + "\nParameters: INDEX "
+            + PREFIX_CODE + "MOD_CODE "
+            + PREFIX_NAME + "EXAM_NAME "
+            + PREFIX_DAY + "DAY "
+            + PREFIX_START + "START_TIME "
+            + PREFIX_END + "END_TIME "
+            + PREFIX_LINK + "LINK "
+            + PREFIX_VENUE + "VENUE "
+            + "\nExample: " + COMMAND_WORD + " exam 1 "
+            + PREFIX_CODE + "CS2103 "
+            + PREFIX_NAME + "Final "
+            + PREFIX_DAY + "02/02/1999 "
+            + PREFIX_START + "10:00 "
+            + PREFIX_END + "11:00 "
+            + PREFIX_LINK + "https://www.youtube.com/watch?v=8mL3L9hN2l4 "
+            + PREFIX_VENUE + "Field";
+    private final Index targetIndex;
+    private final ModuleCode targetModuleCode;
+    private final EditExamDescriptor editExamDescriptor;
+
+    /**
+     * Creates an EditExamCommand to edit the Exam at specified {@code Index} of the specified {@code Module}.
+     */
+    public EditExamCommand(Index targetIndex, ModuleCode targetModuleCode, EditExamDescriptor editExamDescriptor) {
+        requireAllNonNull(targetIndex, targetModuleCode, editExamDescriptor);
```

Comment:
shucks i'm reviewing rohit's pr about removing code from delete exam/lesson (issue #89) and i think you can also remove it since edit exam/lesson is only done in the detail view.

but if you want to do another pr at another time then that's fine too

Reply:
No prob, fixed it in this one